front-end/src/components/CreatePost.tsx

```diff
@@ -4,26 +4,37 @@ import { Label, Modal, ModalBody, ModalFooter, ModalHeader } from 'reactstrap';
 import { Post } from '../types/Post';
 import { Form, Input } from 'reactstrap';
 
-export default function CreatePostComponent() {
-    
+const CreatePostComponent = (props: any) => {
+
     const [title, setTitle] = useState('');
     const [desc, setDesc] = useState('');
     const [contentType, setContentType] = useState('');
     const [content, setContent] = useState('');
     const [categories, setCategories] = useState('');
-    const [visibility, setVisibility] = useState('');
-    const [unlisted, setUnlisted] = useState('');
+    const [visibility, setVisibility] = useState('Public');
+    const [unlisted, setUnlisted] = useState(false);
+
+    function changeVisibility(isChecked: boolean) {
+        if (isChecked) {
+            setVisibility("Friends")
+        }
+        else {
+            setVisibility("Public")
+        }
+    }
 
     function sendPost(e:any) {
-        axios.post(process.env.REACT_APP_API_URL + "/api/author/username/posts/", {
+        let data = {
             title: title,
             description: desc,
             visibility: visibility,
             unlisted: unlisted,
             contentType: contentType,
             content: content,
-            categories: categories
-        })
+            categories: [categories]
+        }
+        
+        axios.post(process.env.REACT_APP_API_URL + "/api/author/" + props.loggedInUser.authorId + "/posts/", data)
```

Comment:
We could have some sort of basic UI in the case that a post fails for some reason, kind of like this:  https://github.com/CMPUT404W21T02/group-project/blob/ba3760e021cff78cc8eb5944760e21d5885f8a3f/front-end/src/components/LoginForm.jsx#L63.

What are your thoughts?

Reply:
Yup sounds good, I've added an Alert similarly